### PR TITLE
Update few Orchestrator APIs to include ValidatingType EF queries. 

### PR DESCRIPTION
**Pull request by cristinamanum**
PackageValidation and ValidatorStatuses tables contain data for packages and symbolpackages.
A symbolpacakge and a package could have the same key. 

In this change few APIs are updated  to use validating type filter in order to keep the intended behavior.

### CODE REVIEW

File: src/Validation.Common.Job/Storage/IValidatorStateService.cs [hunk 1]
```diff
@@ -31,17 +31,9 @@ public interface IValidatorStateService
...
-        Task<bool> IsRevalidationRequestAsync(IValidationRequest request);
-
-        /// <summary>
-        /// Check if the request intends to revalidate a package that has already been validated by <see cref="TValidator"/> by
-        /// a different validation request.
-        /// </summary>
-        /// <param name="packageKey">The package key for the validation request.</param>
-        /// <param name="validationId">The validation id for the validation request.</param>
-        /// <returns>Whether the <see cref="TValidator"/> has already validated this request's package in a different validation request.</returns>
-        Task<bool> IsRevalidationRequestAsync(int packageKey, Guid validationId);
```
Comment (shishirx34) [inline src/Validation.Common.Job/Storage/IValidatorStateService.cs]: why is this removed?

Comment (cristinamanum) [reply src/Validation.Common.Job/Storage/IValidatorStateService.cs]: It is not used.
